cpp/Glacier2/callback/SimpleWakeUpService.cpp

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) ZeroC, Inc.
+
+#include "SimpleWakeUpService.h"
+
+#include <algorithm>
+#include <iostream>
+
+using namespace EarlyRiser;
+using namespace std;
+
+namespace
+{
+    // Converts a time stamp to a time point.
+    chrono::system_clock::time_point toTimePoint(int64_t timeStamp)
+    {
+        const int daysBeforeEpoch = 719162;
+
+        chrono::microseconds timePointMicro{timeStamp / 10}; // timeStamp is in ticks (100 nanoseconds)
+        return chrono::system_clock::time_point{timePointMicro - daysBeforeEpoch * 24h};
+    }
+}
+
+Server::SimpleWakeUpService::~SimpleWakeUpService()
+{
+    // Waits for all outstanding tasks to complete.
+    for (auto& task : _tasks)
+    {
+        task.wait();
+    }
+}
+
+void
+Server::SimpleWakeUpService::wakeMeUp(optional<AlarmClockPrx> alarmClock, int64_t timeStamp, const Ice::Current&)
+{
+    if (!alarmClock)
+    {
+        // Should never happen, but in case it does, the Ice runtime will send an Ice::UnknownException to the client.
+        throw std::invalid_argument{"alarmClock is null"};
+    }
+
+    // With C++20, we'll be able to print the time point easily--but not with C++17.
```

Comment:
I would remove this comment, it unrelated to the Ice example.

Reply:
Ok, I'll update Ice/callback & Ice/bidir as well.